src/lib/translators/mongo.ts

```diff
@@ -595,24 +595,58 @@ function transformArgmaxArgmin(step: Readonly<S.ArgmaxStep> | Readonly<S.ArgminS
 /** transform a 'cumsum' step into corresponding mongo steps */
 function transformCumSum(step: Readonly<S.CumSumStep>): MongoStep {
   const groupby = step.groupby ?? [];
+
+  // For retrocompatibility with old configurations
+  if (step.valueColumn) {
+    return [
+      { $sort: { [step.referenceColumn]: 1 } },
+      {
+        $group: {
+          _id: step.groupby ? columnMap(groupby) : null,
+          [step.valueColumn]: { $push: $$(step.valueColumn) },
+          _vqbArray: { $push: '$$ROOT' },
+        },
+      },
+      { $unwind: { path: '$_vqbArray', includeArrayIndex: '_VQB_INDEX' } },
+      {
+        $project: {
+          ...Object.fromEntries(groupby.map(col => [col, `$_id.${col}`])),
+          [step.newColumn ?? `${step.valueColumn}_CUMSUM`]: {
+            $sum: {
+              $slice: [$$(step.valueColumn), { $add: ['$_VQB_INDEX', 1] }],
+            },
+          },
+          _vqbArray: 1,
+        },
+      },
+      { $replaceRoot: { newRoot: { $mergeObjects: ['$_vqbArray', '$$ROOT'] } } },
+      { $project: { _vqbArray: 0 } },
+    ];
```

Comment:
That'll definitly work, but you'll have an easier time converting the `step.valueColumn` into the new syntax and passing into the same logic after.
Something like you did when reading the python pipeline: https://github.com/ToucanToco/weaverbird/pull/999/files#diff-2fddffd5011b87bd7d07cfcd8ab0a50c9eab64afcbeb1a9c90b3fb216671e5a5R18-R20

Unit tests will ensure both cases results are coherent :)